docs/GettingStarted.md

```diff
@@ -2466,3 +2469,24 @@ However, additional instrumentation provided by Datadog can be activated alongsi
 | `OpenTracing::FORMAT_TEXT_MAP` | Yes        |                        |
 | `OpenTracing::FORMAT_RACK`     | Yes        | Because of the loss of resolution in the Rack format, please note that baggage items with names containing either upper case characters or `-` will be converted to lower case and `_` in a round-trip respectively. We recommend avoiding these characters or accommodating accordingly on the receiving end. |
 | `OpenTracing::FORMAT_BINARY`   | No         |                        |
+
+## Common issues
+
+### Payload too large
+
+If you suspect that not all traces from your application are being captured, this might be due to traces that are too large to be processed.
+
+If your application creates long running traces, measured in minutes, or traces with a thousands of operations, such traces can go over our processing limit.
+You can enable [debug mode](#tracer-settings) and check if messages similar to `"Dropping trace. Payload too large"` are being emitted. Keep in mind that debug mode is very verbose. You can also inspect the [Datadog agent logs](https://docs.datadoghq.com/agent/guide/agent-log-files/) for similar messages.
+
+For cases where the payload is too large, you have the option to enable [partial_flush](#tracer-settings) in the tracer. This will break down large traces into smaller chunks that get streamed to Datadog. The side-effect of partial flushing is that service metrics will count a single large trace multiple times: once per chunk streamed.
+
+### Stack level too deep
+
+`ddtrace` uses [Module#prepend](https://ruby-doc.org/core-3.0.0/Module.html#method-i-prepend) when instrumenting gems, as this approach allows for non-competing overriding of the same method by different parties.
+
+A few libraries use [alias](https://ruby-doc.org/core-3.0.0/doc/syntax/miscellaneous_rdoc.html#label-alias) and [Module#alias_method](https://ruby-doc.org/core-3.0.0/Module.html#method-i-alias_method) for overriding methods, which destructively replaces the original method being overridden. This approach is also incompatible with `Module#prepend` overriding, causing, for example, `SystemStackError`.
+
+There's no simple solution if you are seeing `SystemStackError` being raised, but a few libraries have known workarounds:
+
+* `rack-mini-profiler`: [Net::HTTP stack level too deep errors](https://github.com/MiniProfiler/rack-mini-profiler#nethttp-stack-level-too-deep-errors).
```

Comment:
A few suggestions for this:

* Should we ask users to always report us when they find these kinds of issues, so that we can look into them?
* Perhaps we could link every known item in the list with an issue ticket on our side. This way, customers have a centralized location where they can see more information, as well as add comments/ask questions if need be.

Finally, I won't block this PR with this, but I'm convinced we can do better than just document this issue for customers for the issues we already know of.

For instance, the `rack-mini-profiler` patch can be detected:

```ruby
require 'net/http'
require 'rack-mini-profiler'

example = Net::HTTP.allocate
puts example.method(:request).source_location.inspect
```

Outputs:

```bash
$ ruby detect-rack-mini-profiler.rb
["/Users/ivo.anjo/.rvm/gems/ruby-2.7.2/gems/rack-mini-profiler-2.3.1/lib/patches/net_patches.rb", 16
```

Can we go from detection to avoiding the load of our patches when we find that they're going to conflict? Even if it doesn't work in every case, I think catching the few we can and presenting a nice error message rather than breaking the customer's app seems valuable and builds trust in our product.